pywhy_stats/independence/kci.py

```diff
@@ -153,45 +219,29 @@ def _kernel_test(
     null_sample_size: int,
     threshold: float,
     normalize_data: bool,
+    centered: bool,
+    n_jobs: Optional[int],
 ) -> Tuple[float, float]:
-    if X.ndim == 1:
-        X = X.reshape(-1, 1)
-    if Y.ndim == 1:
-        Y = Y.reshape(-1, 1)
-    if Z is not None and Z.ndim == 1:
-        Z = Z.reshape(-1, 1)
-
-    if normalize_data:
-        # first normalize the data to have zero mean and unit variance
-        # along the columns of the data
-        X = _normalize_data(X)
-        Y = _normalize_data(Y)
-        if Z is not None:
-            Z = _normalize_data(Z)
-
-    if kernel_X is None:
-        kernel_X = _get_default_kernel(X)
-    if kernel_Y is None:
-        kernel_Y = _get_default_kernel(Y)
-
-    Kx = kernel_X(X)
-    Ky = kernel_Y(Y)
+    X, Y, Z = _preprocess_kernel_data(X, Y, Z, normalize_data)
+
+    # if Z is not None:
+    # concatenate the (X, Z) data to compute the K_xz kernel
+    # X = np.concatenate((X, Z), axis=1)
```

Comment:
@bloebp if you uncomment these lines, the unit-test does not fail. However, the lines:

```
# Equivalent to concatenating and then estimating the kernel matrix
        Kx *= Kz  # type: ignore
```

makes the unit-test fail. Wanna take a look? I'm not sure why, but I'm also not sure when the multiplication trick is valid?